### PR TITLE
More cleanup: drop support for Clojure < 1.6 (merge after #52)

### PR DESCRIPTION
**Pull request by borkdude**
This drops support for Clojure versions below 1.6. I think that's not unreasonable since that version is already more than 6 years old.

This is a follow up to PR #52.

### CODE REVIEW

**Comment by jafingerhut** [comment]
These changes also look reasonable to me, and agree that continuing to support Clojure < 1.6 seems not so important right now.

**Comment by slipset** [comment]
This now has a conflict.

File: src/flatland/ordered/map.clj [hunk 1]
```diff
@@ -21,15 +19,7 @@
                          )
            (java.util Map Map$Entry)))
 
-;; We could use compile-if technique here, but hoping to avoid
```
Comment (slipset) [inline src/flatland/ordered/map.clj:24]: Could you please explain to me what this has to do with Clojure < 1.6? I'm not familiar at all and would like to learn

File: src/flatland/ordered/map.clj [hunk 2]
```diff
@@ -133,7 +123,7 @@
     (APersistentMap/mapHash this))
   IHashEq
   (hasheq [this]
-    (hasheq-ordered-map this))
+    (hash-unordered-coll this))
```
Comment (slipset) [inline src/flatland/ordered/map.clj:126]: Also, how is this related to Clojure < 1.6 ?

File: src/flatland/ordered/map.clj (hunk 1 above)
Comment (borkdude) [reply src/flatland/ordered/map.clj:25]: @slipset The implementation fell back on `hash-unordered-coll` when it existed and something else if that didn't exist. `hash-unordered-coll` was introduced in Clojure 1.6.

File: src/flatland/ordered/map.clj (hunk 2 above)
Comment (borkdude) [reply src/flatland/ordered/map.clj:126]: See other comment.

**Comment by borkdude** [comment]
Conflict solved, questions answered.
